### DISCUSSION TITLE
Adding Docusaurus Sidebar to custom pages.

### DISCUSSION THREAD

**Discussion by KulkiratSingh** [facebook/docusaurus#4850] [Q&A]
Hi, 

I have a requirement that requires using a sidebar on a custom page. Is it possible to add the docusaurus's [DocSidebar](https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx) to a custom-built page?

Thank You.

**Comment by slorber** [comment]
This is not supported by design (modular), regular pages know nothing about the docs plugin and can't display a sidebar.

A doc must be referenced in the a sidebar to have the sidebar displayed.

Eventually we'll add the ability to force rendering of a sitebar without adding the doc to a sidebar: https://github.com/facebook/docusaurus/discussions/4728#discussioncomment-703440

In any case, you will need to create a doc (not a page). For displaying a sidebar on pages, you would have to implement this yourself using React, and hardcode the sidebar data (because pages don't have access to docs sidebars data by design)
